### PR TITLE
[master] Header removal is not eviction

### PR DESCRIPTION
**Pull request by chrisdennis**

### CODE REVIEW

File: clustered/server/ehcache-service/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainMap.java [hunk 1]
```diff
@@ -354,12 +354,24 @@ public Lock writeLock() {
...
   }
 
-  static class HeadMap<K> extends EvictionListeningReadWriteLockedOffHeapClockCache<K, InternalChain> {
+  public static class HeadMap<K> extends EvictionListeningReadWriteLockedOffHeapClockCache<K, InternalChain> {
 
     public HeadMap(EvictionListener<K, InternalChain> listener, PageSource source, ChainStorageEngine<K> chainStorage) {
       super(listener, source, chainStorage);
     }
 
+    public void removeAtSlot(int slot, boolean shrink) {
```
Comment (cljohnso) [inline clustered/server/ehcache-service/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainMap.java:363]: No _close_ unit test covers this.  More specifically, there is no unit test in this package that covers `OffHeapChainStorageEngine.AttachedInternalChain.removeHeader` from which this (`removeAtSlot`) method is called.  You might consider adding additional coverage for testing of `AttachedInternalChain` so this method actually gets hit.

**Review by cljohnso — APPROVED** [review]
Approving -- with some unit test suggestions.

File: clustered/server/ehcache-service/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainMap.java [hunk 2]
```diff
@@ -354,12 +354,24 @@ public Lock writeLock() {
   protected void storageEngineFailure(Object failure) {
   }
 
-  static class HeadMap<K> extends EvictionListeningReadWriteLockedOffHeapClockCache<K, InternalChain> {
+  public static class HeadMap<K> extends EvictionListeningReadWriteLockedOffHeapClockCache<K, InternalChain> {
```
[the remark this answers is not in the document]
Comment (chrisdennis) [reply clustered/server/ehcache-service/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainMap.java:357]: The only code using it in this project. It's accessed from dependents. Changing the type of the owner reference in the chain storage engine makes this class part of the (internal) API.

File: clustered/server/ehcache-service/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainMap.java (hunk 1 above)
Comment (chrisdennis) [reply clustered/server/ehcache-service/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainMap.java:363]: Will add an appropriate test.

**Review by cljohnso — APPROVED** [review]

**Pull request merged**